src/BackgroundTasks/Task/Job.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace ILIAS\BackgroundTasks\Task;
+
+use ILIAS\BackgroundTasks\IO;
+use ILIAS\BackgroundTasks\Task;
+
+/**
+ * Interface Job
+ *
+ * @package ILIAS\BackgroundTasks\Task
+ *
+ *          A Task, which can be run without any interaction with the user such as zipping files or just collecting some data
+ */
+interface Job extends Task {
+
+	/**
+	 * @param \ILIAS\BackgroundTasks\IO $input
+	 * @return \ILIAS\BackgroundTasks\IO
+	 */
+	public function run(IO $input);
+
+
+	/**
+	 * @return bool Returns true iff the job supports giving feedback about the percentage done.
+	 */
+	public function supportsPercentage();
```

Comment:
Concerning the UI it will look nicer if you can display a percentage independent loading bar instead of just jumping from 0 to 100.